tests/test_pointing.py

```diff
@@ -0,0 +1,193 @@
+                    g = cmath.rect(g, cmplx)
+                    gains.append(np.array(g))
+                gains = np.array(gains).T
+                bp_gains[i][f][pol] = gains
+
+    return bp_gains
+
+
+bp_gains = generate_bp_gains(offsets, ants, channel_freqs, POLS)
+data_points = pointing.get_offset_gains(
+    bp_gains,
+    just_gains,
+    offsets,
+    ants,
+    TRACK_DURATION,
+    CENTRE_FREQ,
+    BANDWIDTH,
+    N_CHANNELS,
+    POLS,
+    NUM_CHUNKS)
+beams = pointing.beam_fit(data_points, ants, NUM_CHUNKS)
+pointing_offsets = pointing.calc_pointing_offsets(
+    ants,
+    MIDDLE_TIME,
+    TEMPERATURE,
+    HUMIDITY,
+    PRESSURE,
+    beams,
+    target,
+    az_el_adjust)
+
+
+# Test that length of data_points equals the legnth of antenna list
+def test_get_offset_gains_len():
+    assert len(data_points) == len(ants)
+
+# Test that incorrect shape of bp_gains will raise IncorrectShape Error
+
+
+def test_get_offset_gains_shape():
+    with pytest.raises(ValueError):
+        pointing.get_offset_gains(
+            bp_gains[0],
+            just_gains[0],
+            offsets,
+            ants,
+            TRACK_DURATION,
+            CENTRE_FREQ,
+            BANDWIDTH,
+            N_CHANNELS,
+            POLS,
+            NUM_CHUNKS)
+
+# Test legnth of each data_points element
+
+
+def test_get_offset_gains_len2():
+    for i in range(0, len(data_points)):
+        assert len(list(data_points.items())[i][1]) == NUM_CHUNKS * len(offsets)
+        for j in range(0, NUM_CHUNKS * len(offsets)):
+            assert len(list(data_points.items())[i][1][j]) == 5
+
+# Testing that the output of beam_fit are of type BeamPatternFit
+
+
+def test_beam_fit_type():
+    assert len(beams) == len(ants)
+    for i in ants:
+        assert type(beams[i.name][0]) and isinstance(beams[i.name][-1], type(None))
+        for j in range(1, NUM_CHUNKS - 1):
+            assert isinstance(beams[i.name][j], pointing.BeamPatternFit)
+
+# Multiple small type errors for calc_pointing_offsets
+
+# Test that the legnth of each pointing offset solution =10 (5 sets of (x,y) coordinates)
+
+
+def test_calc_pointing_offsets_len():
+    assert len(pointing_offsets) == len(ants)
+    for i in range(0, len(pointing_offsets)):
+        assert len(list(pointing_offsets.items())[i][1]) == 10
+
+# Compare widths of simulated primary beams from beam_fit and original beam object
+
+
+def test_fit_primary_beams():
+    future_ex_widths = {}
+    for a, ant in enumerate(ants):
+        ex_width = []
+
+        for chunk in range(0, NUM_CHUNKS):
+            expected_width = pointing._voltage_beamwidths(
+                ant.beamwidth, chunk_freqs[chunk], ant.diameter)
+            ex_width.append(expected_width)
+            future_ex_widths[ant.name] = ex_width
+
+    # Feeding simulated data_points into beam_fit function
+    beams = pointing.beam_fit(data_points, ants, NUM_CHUNKS)
+    # Comparing output of beam_fit to the original beam object
```

Comment:
I think the most important test by far is whether we found the expected offset / beam centre, since that's what pointing is all about.